lib/elixir/lib/enum.ex

```diff
@@ -2494,9 +2499,17 @@ defmodule Enum do
   defp enum_to_string(entry) when is_binary(entry), do: entry
   defp enum_to_string(entry), do: String.Chars.to_string(entry)
 
-  defp random_index(n) do
-    :rand.uniform(n + 1) - 1
-  end
+  @spec random_integer(integer, integer) :: integer
```

Comment:
We actually do in a few places:
```
$ ag "@spec [^\n]+\s+defp"

lib/elixir/lib/dict.ex
364:  @spec unsupported_dict(t) :: no_return
365:  defp unsupported_dict(dict) do

lib/elixir/lib/macro.ex
79:  @spec binary_op_props(atom) :: {:left | :right, precedence :: integer}
80:  defp binary_op_props(o) do

lib/elixir/lib/inspect/algebra.ex
537:  @spec fits?(integer, [{integer, mode, t}]) :: boolean
538:  defp fits?(w, _) when w < 0,                      do: false
549:  @spec format(integer | :infinity, integer, [{integer, mode, t}]) :: [binary]
550:  defp format(_, _, []),                                do: []
```

I remember I asked this a while ago to José, and he mentioned it's OK since it can useful for Dialyzer.